Makefile

```diff
@@ -526,21 +565,21 @@ ifdef ROCM_BASE
 	@echo -n '$(ROCM_BASE)/bin:'                                            >> $@
 endif
 ifneq ($(SYCL_BASE),)
-ifeq ($(wildcard $(ONEAPI_ENV)),)
-	@echo -n '$(SYCL_BASE)/bin:'                                            >> $@
-endif
+	@echo -n '$(SYCL_PATH):'                                                >> $@
 endif
 	@echo '$$PATH'                                                          >> $@
-# check if oneAPI environment file exists
-ifneq ($(wildcard $(ONEAPI_ENV)),)
-	@echo 'source $(ONEAPI_ENV)'                                            >> $@
-else
+ifneq ($(SYCL_BASE),)
+	@# load the CPU OpenCL runtime
 	@echo 'export OCL_ICD_FILENAMES=$(OCL_ICD_FILENAMES)'                   >> $@
+	@# enable double precision floating point emulation for Intel GPUs
+	@# see https://github.com/intel/compute-runtime/blob/master/opencl/doc/FAQ.md#feature-double-precision-emulation-fp64
+	@echo 'export IGC_EnableDPEmulation=1'                                  >> $@
+	@echo 'export OverrideDefaultFP64Settings=1'                            >> $@
 endif
 
 define TARGET_template
 $(1): $$(foreach dep,$$($(1)_EXTERNAL_DEPENDS),$$($$(dep)_DEPS)) | $(DATA_DEPS)
-	+$(MAKE) -C src/$(1)
+	+$(MAKE) -C src/$(1) TARGET_NAME=$(firstword $(TARGET_NAME) $(1))
```

Comment:
' Neat! Then I'd suggest to
* extend the support to all the `src/*` programs (maybe in a subsequent PR?)
* remove the `TARGET_NAME` definition from the `src/*/Makefile` since it is now always delivered by the top-level `Makefile` (could be done for `sycl` in this PR)
* document this feature in the README

Reply:
Done !